File: ReadLater/ClientApp/src/services/pocket-service.ts
import RequestService from './request-service';

class PocketService extends RequestService {
    constructor() {
        super();
    }

    public async getRequestToken(): Promise<string> {
        var token = await this.requestWithResponse('/api/pocket/token/request', 'post', undefined);
        return token;
    }

    public async getAuthorizeUrl(token: string): Promise<string> {
        var redirectUri = window.location.href;
        return `https://getpocket.com/auth/authorize?request_token=${token}&redirect_uri=${redirectUri}`;
    }

    public async getAccessToken(code: string): Promise<string> {
        var token = await this.requestWithResponse('/api/pocket/token', 'post', undefined);
        return token;
    }
}

const pocketService = new PocketService();

export default pocketService;

File: ReadLater/Services/Pocket/PocketService.cs
using Newtonsoft.Json;
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace ReadLater.Services.Pocket
{
    public class PocketService : IPocketService
    {
        private HttpClient _client;
        private const string ConsumerKey = "84468-64a87fb1c2e843d4d28f1981";

        public PocketService()
        {
            _client = new HttpClient();
            _client.BaseAddress = new Uri("https://getpocket.com/v3");
        }

        public async Task<string> GetRequestTokenAsync()
        {
            var content = new
            {
                consumer_key = ConsumerKey,
                redirect_uri = "http://localhost:5001/"
            };

            var json = JsonConvert.SerializeObject(content);
            var request = new HttpRequestMessage(HttpMethod.Post, "/v3/oauth/request");
            request.Headers.Add("X-Accept", "application/json");
            request.Content = new StringContent(json.ToString(), Encoding.UTF8, "application/json");

            var response = await _client.SendAsync(request);
            var responseContent = await response.Content.ReadAsStringAsync();
            var token = JsonConvert.DeserializeObject<RequestToken>(responseContent);

            return token.Code;
        }

        public async Task<string> GetAccessTokenAsync(string code)
        {
            var content = new
            {
                consumer_key = ConsumerKey,
                code
            };

            var json = JsonConvert.SerializeObject(content);
            var request = new HttpRequestMessage(HttpMethod.Post, "/v3/oauth/authorize");
            request.Headers.Add("X-Accept", "application/json");
            request.Content = new StringContent(json.ToString(), Encoding.UTF8, "application/json");

            var response = await _client.SendAsync(request);
            var responseContent = await response.Content.ReadAsStringAsync();
            var token = JsonConvert.DeserializeObject<AccessToken>(responseContent);

            return token.Token;
        }
    }
}
